onvif/src/discovery/mod.rs

```diff
@@ -150,22 +151,37 @@ async fn get_responding_addr<F, Fut>(
     mut check_addr: F,
 ) -> Option<Device>
 where
-    F: FnMut(Url) -> Fut,
+    F: FnMut(Url) -> Fut + Copy,
```

Comment:
I don't think `FnMut + Copy` makes much sense. If a closure implements only `FnMut` but not `Fn`, I'm pretty sure it can't be `Copy`. Let's make this `Fn(Utl) -> Fut + Copy` and drop the `mut` from `check_addr`.

Reply:
Or really `+ Clone`, if possible.

Reply:
Complains about `Fn(Url) -> Fut + Clone`:

```
error[E0507]: cannot move out of `check_addr`, a captured variable in an `FnMut` closure
   --> onvif/src/discovery/mod.rs:171:39
    |
151 |       check_addr: F,
    |       ---------- captured outer variable
...
171 |                   .map(|url| async move {
    |  _______________________________________^
172 | |                     check_addr(url.clone()).await.then(|| {
    | |                     ----------
    | |                     |
    | |                     move occurs because `check_addr` has type `F`, which does not implement the `Copy` trait
    | |                     move occurs due to use in generator
173 | |                         debug!("Responding addr: {:?}", url);
174 | |                         Device {
...   |
178 | |                     })
179 | |                 })
    | |_________________^ move out of `check_addr` occurs here
```

Reply:
Yeah you'd have to actually `.clone()` check_addr, probably before constructing the `async move {}` block but inside the map closure. But I guess it's okay to leave it as `Fn(Url) -> Fut + Copy`.